plugins/transport/socket/main.go

```diff
@@ -96,6 +104,20 @@ func (s *Socket) Run(ctx context.Context, w transport.WriteFn, done chan bool) {
 				return
 			}
 
+			// whole buffer was used, so we are potentially handling larger message
+			// and have to increase buffer size
+			if n == len(msgBuffer) {
+				s.logger.Warnf("insufficient buffer size: %dB", len(msgBuffer))
+				newSize := int64(len(msgBuffer)) + initBuffSize
+				if newSize <= maxBuffSize {
+					msgBuffer = make([]byte, newSize)
+					s.logger.Infof("increased buffer size: %dB", newSize)
+				} else {
+					s.logger.Warnf("reading buffer insufficient, but cannot increase size anymore")
```

Comment:
what issues arise if you try to increase the size more than once?

Reply:
More than three times. None besides reading large messages which means performance impact.